runtime/caches/redis.ts

```diff
@@ -1,36 +1,37 @@
-import { Redis } from "https://deno.land/x/upstash_redis@v1.22.1/mod.ts";
-
-import { logger, tracer } from "../../observability/otel/config.ts";
 import {
   assertCanBeCached,
   assertNoOptions,
   withCacheNamespace,
 } from "./common.ts";
+import { createClient } from "npm:@redis/client@^1.6.0";
+import { logger } from "../../observability/mod.ts";
 
-const redisUrl = Deno.env.get("UPSTASH_REDIS_REST_URL");
-const redisToken = Deno.env.get("UPSTASH_REDIS_REST_TOKEN");
-
-export const redis = redisUrl && redisToken
-  ? new Redis({
-    url: redisUrl,
-    token: redisToken,
-    enableTelemetry: true,
-  })
-  : null;
-
-interface ResponseMetadata {
-  body: string;
-  status: number;
-  headers: [string, string][];
-}
+const REDIS_URL = Deno.env.get("REDIS_URL") || "redis://localhost:6379/0";
+
+const redis = createClient({
+  url: REDIS_URL,
+}).on("error", (err: Error) => {
+  logger.error(err);
```

Comment:
if you just output the error using `logger.error` so it becomes difficult to find them, you can use the second parameter which is `Record<string, string>` to pass additional info. you can use 
```json
{
type: "cache_error",
engine: "redis",
operation: "createclient"
}
```

same for others